src/interpreter_funcs/voxel_boolean_union.rs

```diff
@@ -0,0 +1,261 @@
+                }),
+                optional: false,
+            },
+            ParamInfo {
+                name: "Detailed Mesh Analysis",
+                description: "Reports detailed analytic information on the created mesh.\n\
+                              The analysis may be slow, therefore it is by default off.",
+                refinement: ParamRefinement::Boolean(BooleanParamRefinement {
+                    default_value: false,
+                }),
+                optional: false,
+            },
+        ]
+    }
+
+    fn return_ty(&self) -> Ty {
+        Ty::Mesh
+    }
+
+    fn call(
+        &mut self,
+        args: &[Value],
+        log: &mut dyn FnMut(LogMessage),
+    ) -> Result<Value, FuncError> {
+        let mesh1 = args[0].unwrap_mesh();
+        let mesh2 = args[1].unwrap_mesh();
+        let voxel_dimensions = Vector3::from(args[1].unwrap_float3());
+        let growth_u32 = args[3].unwrap_uint();
+        let growth_f32 = growth_u32 as f32;
+        let fill = args[4].unwrap_boolean();
+        let error_if_large = args[5].unwrap_boolean();
+        let analyze_bbox = args[6].unwrap_boolean();
+        let analyze_mesh = args[7].unwrap_boolean();
+
+        if voxel_dimensions
+            .iter()
+            .any(|dimension| approx::relative_eq!(*dimension, 0.0))
+        {
+            let error = FuncError::new(FuncBooleanUnionError::VoxelDimensionsZero);
+            log(LogMessage::error(format!("Error: {}", error)));
+            return Err(error);
+        }
+
+        let bbox1 = mesh1.bounding_box();
+        let bbox2 = mesh2.bounding_box();
+        let bbox =
+            BoundingBox::union([bbox1, bbox2].iter().copied()).expect("Failed to create union box");
+        let voxel_count = voxel_cloud::evaluate_voxel_count(&bbox, &voxel_dimensions);
+
+        log(LogMessage::info(format!("Voxel count = {}", voxel_count)));
+
+        if error_if_large && voxel_count > VOXEL_COUNT_THRESHOLD {
+            let suggested_voxel_size =
+                voxel_cloud::suggest_voxel_size_to_fit_bbox_within_voxel_count(
+                    voxel_count,
+                    &voxel_dimensions,
+                    VOXEL_COUNT_THRESHOLD,
+                );
+
+            let error = FuncError::new(FuncBooleanUnionError::TooManyVoxels(
+                VOXEL_COUNT_THRESHOLD,
+                suggested_voxel_size.x,
+                suggested_voxel_size.y,
+                suggested_voxel_size.z,
+            ));
+            log(LogMessage::error(format!("Error: {}", error)));
+            return Err(error);
+        }
+
+        let mut voxel_cloud1 = ScalarField::from_mesh(mesh1, &voxel_dimensions, 0.0, growth_u32);
+        let mut voxel_cloud2 = ScalarField::from_mesh(mesh2, &voxel_dimensions, 0.0, growth_u32);
+
+        voxel_cloud1.compute_distance_field(&(0.0..=0.0));
+        voxel_cloud2.compute_distance_field(&(0.0..=0.0));
+
+        let boolean_union_range = &(-growth_f32..=growth_f32);
```

Comment:
Maybe I am missing something, but why is this different from the other 2 operations? Shouldn't the `meshing_range` be ok provided that `ScalarField::compute_distance_field` correctly computes what is inside and what is outside?

Reply:
yes, maybe it's even more correct because using the boolean union range ignores the insides of the second scalar field